lib/ex_gram/adapter/tesla.ex

```diff
@@ -111,7 +113,29 @@ if Code.ensure_loaded?(Tesla) do
     defp opts(), do: [adapter: adapter_opts()]
     defp adapter_opts(), do: [connect_timeout: 5_000, timeout: 60_000, recv_timeout: 60_000]
 
-    defp custom_middlewares(),
-      do: Application.get_env(:ex_gram, __MODULE__, [])[:middlewares] || []
+    defp format_middleware({m, f, a}) do
+      case apply(m, f, a) do
+        {_, _} = middleware -> {:ok, middleware}
+        _ -> :error
+      end
+    end
+
+    defp format_middleware({_, _} = mf), do: mf
+    defp format_middleware(_), do: :error
+
+    defp custom_middlewares() do
+      (Application.get_env(:ex_gram, __MODULE__, [])[:middlewares] || [])
```

Comment:
Save this to a variable, I think having parenthesis before a pipe doesn't look good